src/c_api/c_api.cc

```diff
@@ -177,10 +177,39 @@ void CustomFComputeDispatcher(const std::string op_name,
     return static_cast<void*>((*gpualloc)(size));
   };
 
+  // MXNet random number generator lambda to be invoked by custom library
+  // custom library will use gpu rng whenever possible
+  auto rng_caller = [&](RandGenType rand_type, int seed) {
+#if MXNET_USE_CUDA
```

Comment:
Some users may use MXNet built with `MXNET_USE_CUDA=1` on the machine with only CPU. In this case, the GPU random number generator is not available.

It is better to decide which device to choose in the user-side code, namely the code of custom operators.

Reply:
thanks for the timely review. we can let user pass in a "cpu" or "gpu" string to decide it, just like the forward/backward function

Also I am just wondering if we should make a default choice of device?

Reply:
Agree to making a default choice : )

I think it should depend on the context of custom operator, rather that the version (GPU or only CPU) of MXNet. 

If users build a CPU custom operator and use the GPU version of MXNet, on a machine with only CPU, it still work since the CPU random number generator is used.

Reply:
I dont think we should have a default, user should be explicit which one they want to use. otherwise we introduce possible confusion